src/unix/process.c

```diff
@@ -40,7 +40,7 @@
 extern char **environ;
 #endif
 
-#ifdef __linux__
+#if defined(__linux__) || (defined(__FreeBSD_kernel__) && defined(__GLIBC__))
```

Comment:
Similar to my other comment, I think grp.h is not something specific to the FreeBSD kernel, but GNU libc should always provide it.
`#if defined(__linux__) || defined(__GLIBC__)`
should be valid for GNU/Hurd, GNU/anything else.